httptransport/server.go

```diff
@@ -161,26 +161,28 @@ func (t *Server) configureComboMode(ctx context.Context) error {
 // ConfigureIndexerMode configures the HttpTransport for IndexerMode.
 //
 // This mode runs only an Indexer in a single process.
-func (t *Server) configureIndexerMode(_ context.Context) error {
+func (t *Server) configureIndexerMode(ctx context.Context) error {
 	// requires only indexer service
 	if t.indexer == nil {
 		return clairerror.ErrNotInitialized{Msg: "IndexerMode requires an indexer service"}
 	}
+	prefix := indexerRoot + apiRoot
 
-	ratelimitMW := rate.NewRateLimitMiddleware(t.conf.Indexer.IndexReportRequestConcurrency)
-
-	t.Handle(AffectedManifestAPIPath,
-		intromw.InstrumentedHandler(AffectedManifestAPIPath, t.traceOpt, AffectedManifestHandler(t.indexer)))
-
-	t.Handle(IndexAPIPath,
-		intromw.InstrumentedHandler(IndexAPIPath, t.traceOpt, ratelimitMW.Handler(IndexAPIPath, IndexHandler(t.indexer))))
-
-	t.Handle(IndexReportAPIPath,
-		intromw.InstrumentedHandler(IndexReportAPIPath+"GET", t.traceOpt, IndexReportHandler(t.indexer)))
-
-	t.Handle(IndexStateAPIPath,
-		intromw.InstrumentedHandler(IndexStateAPIPath, t.traceOpt, IndexStateHandler(t.indexer)))
-
+	v1, err := NewIndexerV1(ctx, prefix, t.indexer, t.traceOpt)
+	if err != nil {
+		return fmt.Errorf("indexer configuration: %w", err)
+	}
+	sem := semaphore.NewWeighted(int64(t.conf.Indexer.IndexReportRequestConcurrency))
```

Comment:
I think if the user leaves `IndexReportRequestConcurrency` undefined in the config every request will be 429ed as opposed to none of them, if I'm reading correctly